### PR TITLE
Add #13: 1.18 support

### PR DESCRIPTION
**Pull request by spnda**
This adds basic 1.18 support. Some features might not work and some features could be added for higher mountains or the deeper caves but this gives a functional base to work on. `BiomeDeepAnalyzer` (BiomeDepthAnalyzer*?) could be removed I think because `Biome#getDepth` was removed and biome generation works differently now. I have, however, left it in and want your comment on that. Otherwise, these are simple changes compared to 1.17.

### CODE REVIEW

File: src/main/java/org/orecruncher/dsurround/processing/BiomeSoundHandler.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
     public static final int MOOD_SOUND_MAX_RANGE = 16;
 
     // Reusable map for biome acoustic work
-    private static final Object2FloatOpenHashMap<ISoundFactory> WORK_MAP = new Object2FloatOpenHashMap<>(8, 1F);
+    private static final Object2FloatOpenHashMap<ISoundFactory> WORK_MAP = new Object2FloatOpenHashMap<>(8, 0.5F);
```
Comment (ChloeDawn) [inline src/main/java/org/orecruncher/dsurround/processing/BiomeSoundHandler.java:29]: Why was this changed to `0.5F`?

Comment (spnda) [reply src/main/java/org/orecruncher/dsurround/processing/BiomeSoundHandler.java:29]: Because the constructor of `Object2FloatOpenHashMap` expects that float to be `> 0,0 && < 1.0` and will throw a Exception when `1.0F` is passed. I will, however, probably replace the constant with a static field from fastutil's `Hash` class, which defines `0.75f` as the default load factor and `0.5f` for a small but quick map.

I don't know why this changed while switching to 1.18 or it mightve been broken before too.

**Comment by Catboi8** [comment]
Works well, tho possible mod incompatibility or something?
Any time i attempt to close a world this happens

```
[12:49:03] [Render thread/INFO]: Saved data for singleplayer-New World (1)
[12:49:03] [Sound engine/ERROR]: Stop: Invalid name parameter.
[12:49:03] [Render thread/INFO]: Stopping worker threads
[12:49:03] [Render thread/FATAL]: Reported exception thrown!
net.minecraft.class_148: mouseClicked event handler
	at net.minecraft.class_437.method_25412(class_437.java:496) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1601(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22686(class_312.java:165) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.execute(class_1255.java:101) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_22684(class_312.java:165) ~[client-intermediary.jar:?]
	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.2.2.jar:build 10]
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3101) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:193) ~[client-intermediary.jar:?]
	at net.minecraft.class_1041.method_15998(class_1041.java:310) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1142) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:733) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:236) [client-intermediary.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:602) [fabric-loader-0.12.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77) [fabric-loader-0.12.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.12.9.jar:?]
Caused by: java.lang.IllegalStateException: Tried to release unknown channel
	at net.minecraft.class_4225.method_19662(class_4225.java:301) ~[client-intermediary.jar:?]
	at net.minecraft.class_4235$class_4236.method_19736(class_4235.java:42) ~[client-intermediary.jar:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at java.util.Collections$SetFromMap.forEach(Collections.java:5700) ~[?:?]
	at net.minecraft.class_4235.method_19728(class_4235.java:93) ~[client-intermediary.jar:?]
	at net.minecraft.class_1140.method_4843(class_1140.java:177) ~[client-intermediary.jar:?]
	at net.minecraft.class_1144.method_4881(class_1144.java:245) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_18098(class_310.java:2229) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_18096(class_310.java:2204) ~[client-intermediary.jar:?]
	at net.minecraft.class_433.method_19836(class_433.java:77) ~[client-intermediary.jar:?]
	at net.minecraft.class_4185.method_25306(class_4185.java:40) ~[client-intermediary.jar:?]
	at net.minecraft.class_4264.method_25348(class_4264.java:16) ~[client-intermediary.jar:?]
	at net.minecraft.class_339.method_25402(class_339.java:121) ~[client-intermediary.jar:?]
	at net.minecraft.class_4069.method_25402(class_4069.java:27) ~[client-intermediary.jar:?]
	at net.minecraft.class_312.method_1611(class_312.java:94) ~[client-intermediary.jar:?]
	at net.minecraft.class_437.method_25412(class_437.java:491) ~[client-intermediary.jar:?]
	... 19 more
```

**Comment by spnda** [comment]
> Works well, tho possible mod incompatibility or something? Any time i attempt to close a world this happens
> 
> ```
> [12:49:03] [Render thread/INFO]: Saved data for singleplayer-New World (1)
> [12:49:03] [Sound engine/ERROR]: Stop: Invalid name parameter.
> [12:49:03] [Render thread/INFO]: Stopping worker threads
> [12:49:03] [Render thread/FATAL]: Reported exception thrown!
> net.minecraft.class_148: mouseClicked event handler
> 	at net.minecraft.class_437.method_25412(class_437.java:496) ~[client-intermediary.jar:?]
> 	...

Do you know if this also happens on 1.17? There was no code changes specifically towards this, DS only increases/decreases the amount of stream/static source channels.

**Comment by Catboi8** [comment]
Upon a second look at the issues it seems to be the same as #10, ill attempt to narrow it down there

**Comment by ThexXTURBOXx** [comment]
Superseded by #17 I guess? :)
